src/main/java/core/basesyntax/model/Operation.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public enum Operation {
+    BALANCE("b"),
+    SUPPLY("s"),
+    PURCHASE("p"),
+    RETURN("r");
+
+    private String code;
+
+    Operation(String code) {
+        this.code = code;
+    }
+
+    public String getCodeOfOperation() {
+        return code;
+    }
+
+    public static Operation getOperationOf(String letter) {
+        return Arrays.stream(Operation.values())
+                .filter(o -> o.getCodeOfOperation().equals(letter))
```

Comment:
please choose a proper name for parameters in the lambdas; other developers will need to spend additional time figuring out what 'o' means

Reply:
still exists.